lib/JitEngine.cpp

```diff
@@ -42,7 +43,11 @@ using namespace llvm;
 Expected<std::unique_ptr<TargetMachine>>
 JitEngine::createTargetMachine(Module &M, StringRef Arch, unsigned OptLevel) {
   Triple TT(M.getTargetTriple());
+#if LLVM_VERSION_MAJOR >= 18
+  std::optional<llvm::CodeGenOptLevel> CGOptLevel = CodeGenOpt::getLevel(OptLevel);
```

Comment:
One could also just use `auto` here

Reply:
Let's go for `auto` to avoid preprocessor pollution

Reply:
Done.